perses/tests/test_relative_point_mutation_setup.py

```diff
@@ -60,4 +83,35 @@ def test_PointMutationExecutor_solvated():
             flatten_torsions=False,
             flatten_exceptions=False,
             conduct_endstate_validation=False
-        )
\ No newline at end of file
+        )
+
+
+def test_PointMutationExecutor_without_counterion():
+    """
+    Check that a PointMutationExecutor can be instantiated properly for ALA->ASP dipeptide in solvent without a counterion.
+
+    """
+    from pkg_resources import resource_filename
+    from simtk import unit
+
+    from perses.app.relative_point_mutation_setup import PointMutationExecutor
+
+    pdb_filename = resource_filename("perses", "data/ala_vacuum.pdb")
+    solvent_delivery = PointMutationExecutor(
+        pdb_filename,
+        "1",
+        "2",
+        "ASP",
+        ionic_strength=0.15 * unit.molar,
+        flatten_torsions=True,
+        flatten_exceptions=True,
+        conduct_endstate_validation=False,
+        transform_waters_into_ions_for_charge_changes=False
+    )
+    htf = solvent_delivery.get_apo_htf()
+
+    # If there is no counterion, there should be no water atoms in the core atom class
+    solvent_atoms = set(htf.hybrid_topology.select('water'))
+    assert len(solvent_atoms.intersection(htf._atom_classes['core_atoms'])) == 0, "There are water atoms in the core atom " \
+                                                                                  "class, which may mean that a counterion is being introduced"
+
```

Comment:
This test is a good one, but I wonder if we also want a test that directly computes and validates the charge of the mutation with and without the `transform_waters_into_ions_for_charge_changes` flag. Would this be easy to accomplish with our current data structures and methods?

Reply:
I can't think of an easy way to do this, but maybe you could open an issue for this and we can discuss at the next dev sync?